src/Transport/Config/ExtensionPoints/AzureServiceBusNamespacePartitioningSettings.cs

```diff
@@ -29,7 +29,7 @@ public AzureServiceBusNamespacePartitioningSettings(SettingsHolder settings)
             return this;
         }
 
-        public AzureServiceBusNamespacePartitioningSettings AddNamespace(string name, string connectionString)
+        public AzureServiceBusNamespacePartitioningSettings AddNamespace(string name, string connectionString, NamespacePurpose purpose = NamespacePurpose.Partitioning)
```

Comment:
I don't think that is a good idea, UseNamespaceNameInsteadOfConnectionString is only temporary, in future releases it will be default. then we'd have to change the API again